### PR TITLE
DL-4449: Corrected incorrect email issue and improved validation

### PR DESCRIPTION
**Pull request by davidmthornton**
DL-4449: Corrected incorrect email send and improved validation

Bug fix

I have corrected a problem where the wrong email was being sent to users which had content for an already trading user with information about penalties involved.

The fix is to ensure that if the user answers yes to the trading before 31st March 2019 question, they will be classed as an old business and will receive the already trading email template. If they answer no they will receive the new trader registration email.

There was also some missing validation where the user could say they were trading before 31st March 2019, but then provide a start trading date which was after that day and if they say no, provide a date before that date. I have put a fix in for this.

I also have an outstanding question about the legislation. It states that the rules changed from 1st April 2016, so the question should most probably read "before 1st April 2016". This will be handled on a future ticket

### CODE REVIEW

File: app/forms/submapping/TupleDateMapping.scala [hunk 1]
```diff
@@ -43,14 +43,25 @@ object TupleDateMapping {
...
-  val yearMustBe4Digits = (fieldKey: String) => (date: TupleDate) => date.year.matches("""^\d{4}$""") match {
+  val yearMustBe4Digits: String => TupleDate => ValidationResult = (fieldKey: String) => (date: TupleDate) => date.year.matches("""^\d{4}$""") match {
     case true => Valid
     case _ => simpleErrorMessage(fieldKey, "awrs.business_details.error.year_toSmall")
   }
 
-  def CompulsoryTupleDateFormatter(isEmptyErrMessage: (String) => Invalid,
-                                   isInvalidErrMessage: (String) => Invalid,
-                                   dateRangeCheck: Option[(String) => (TupleDate) => ValidationResult] = Some(yearMustBe4Digits)) = new Formatter[TupleDate] {
+//  private val cutOffAWBusinessStartDate = "01/04/2016"
```
Comment (paulanderson1234) [inline app/forms/submapping/TupleDateMapping.scala:51]: Are we keeping this for possible future implementation?

File: test/services/ApplicationServiceTest.scala [hunk 2]
```diff
@@ -1188,6 +1188,24 @@ class ApplicationServiceTest extends AwrsUnitTestTraits
         }
       }
     }
+
+    "isNewBusiness" must {
+      "return true if the business is not a NewAWBusiness (already set up and newly trading in alcohol)" in {
```
Comment (paulanderson1234) [inline test/services/ApplicationServiceTest.scala:1193]: Is this description correct - surely returns true if it IS a NewAWBusiness ?

Comment (paulanderson1234) [inline test/services/ApplicationServiceTest.scala:1193]: Not understanding the wording of this - I see this get set to false in the db i.e. before 31 March 2016 but I may be getting confused with the big picture as opposed to a specific page ...

File: app/forms/submapping/TupleDateMapping.scala (hunk 1 above)
Comment (davidmthornton) [reply app/forms/submapping/TupleDateMapping.scala:51]: I've left that there by mistake - I'll remove it.

File: test/services/ApplicationServiceTest.scala (hunk 2 above)
Comment (davidmthornton) [reply test/services/ApplicationServiceTest.scala:1193]: Must have had a brain fart here. I'll update the wording of this.

**Pull request merged**